erpnext/stock/doctype/item/item.py

```diff
@@ -806,6 +807,72 @@ def validate_has_variants(self):
 			if frappe.db.exists("Item", {"variant_of": self.name}):
 				frappe.throw(_("Item has variants."))
 
+	def validate_attributes_in_variants(self):
+		if not self.has_variants:
+			return
+
+		old_doc = self.get_doc_before_save()
+		old_doc_attributes = set([attr.attribute for attr in old_doc.attributes])
+		own_attributes = [attr.attribute for attr in self.attributes]
+
+		# Check if old attributes were removed from the list
+		# Is old_attrs is a subset of new ones
+		# that means we need not check any changes
+		if old_doc_attributes.issubset(set(own_attributes)):
+			return
+
+		from collections import defaultdict
+
+		# get all item variants
+		items = [item["name"] for item in frappe.get_all("Item", {"variant_of": self.name})]
+		items_string = ', '.join(items)
+
+		# fetch all attributes of these items
+		item_attributes = frappe.get_all(
+			"Item Variant Attribute",
+			filters={"parent": ["in", "({})".format(items_string)]},
+			fields=["attribute", "parent"]
+		)
```

Comment:
Since we know some attribute has been deleted at this point from the template, we can just get the deleted attributes via `old_doc_attributes.difference(own_attributes)`. Then just get all item variant attribute details of only deleted attributes
(line 838) Looping through every Item Variant Attribute will have a lot of repetition.

Reply:
@marination I've made the requested changes, can you re-review